### DISCUSSION TITLE
Problem with TensorFlow in RStudio.

### DISCUSSION THREAD

**Discussion by smisani** [teamalgoritma/community#162] [Q&A] [answered]
Dear Algoritma team members,

I'm facing difficulty in reticulating TensorFlow in my RStudio. I've done all steps as prescribed in the manual but to no avail. Appreciate your kind help on this matter.  

Thanks.
SB

![image](https://user-images.githubusercontent.com/103245111/176371302-75da2d7b-f3fe-4104-85f3-77ea8ebbcb88.png)

**Comment by ranydwi** [comment, marked as answer]
Hi, @smisani!

Thank you for the question.
Have you ever tried to install the TensorFlow in R using Github? You could try a few steps below:
1. You need to deactivate (uninstall library Keras and TensorFlow in R) if the packages are already installed
2. `install.packages("devtools")` in console
3. `library (devtools)` in console
4. Reinstall Tensorflow and Keras from Github, here is the code (type in your console one by one)
`devtools::install_github("rstudio/tensorflow")`
`devtools::install_github("rstudio/keras")`
5. Furthermore, if it is successful, you can run the code below to install both of the following (one by one in your console):
`tensorflow::install_tensorflow()`
`tensorflow::tf_config()`
6. If the process already done you need to restart your R studio
Restart R > Close R
7. Reopen R then type again in the console:
```
library (keras)
library (tensorflow)
Model <- keras_model_sequential()
```
Hope this will help you.

**Reply by smisani** [reply]
apparently when I tried installing devtools, my RStudio crashed because it keeps on restarting. May I know what RStudio and R version should I be using to follow the instruction?

Thanks,
SB

**Reply by ranydwi** [reply]
Dear @smisani 

The same situation happened with one of the student in night class. The problem solved with this student when she did several things below:
1. Make sure the `devtools` package is not installed
2. Restart your laptop first and then try to restart R before trying to install `devtools`
3. the R version used is the latest version R-4.2.1

Hope this will help :)

**Reply by smisani** [reply]
Thanks, but I'm still encountering problems when I tried: `tensorflow::install_tensorflow()` 
![image](https://user-images.githubusercontent.com/103245111/176620846-d22dec88-c416-4408-8e49-a6c81f6ba8e6.png)

Any help on this? thanks.

Best,
SB

**Reply by ranydwi** [reply]
Dear Bu @smisani
I think the installation needs version requirements. Please try to run these code below, I hope these one will works
`tensorflow::install_tensorflow(version = "2.0.0")`

Maybe the error comes because tensorflow 2.9 not suitable for python > 3.7

**Reply by smisani** [reply]
still not working. T_T
![image](https://user-images.githubusercontent.com/103245111/176813931-5c39091a-5630-416a-8a6d-2c352c3ec103.png)
do I need to uninstall some programs or any other help with this?

**Reply by ranydwi** [reply]
Dear Bu @smisani,

Huhu... stiil not working ya, but, it's okayy we can try once again :"). Based on the last screenshot you attached, I think you need to follow the step from the first step again. You need to restart R, because the installation of tensorflow need it. Then, you need to uninstall Keras, Tensorflow, and Devtools from the package list. 

I hope that if we try really from the start it will work.
Semangaat Bu! :)

**Reply by onlyphantom** [reply]
Also from the screenshot, the oldest version of tensorflow is 2.2.0, then 2.2.1, ... 2.9.1. 

So `tensorflow::install_tensorflow(version = "2.2.0")` is the lowest you can go.

**Reply by smisani** [reply]
still not working apparently after i tried the `tf_config`.

![image](https://user-images.githubusercontent.com/103245111/176823424-2f11adda-1988-467c-932a-0b8c43d8a68a.png)
T_T
